include/PetriEngine/Colored/VariableVisitor.h

```diff
@@ -289,6 +295,32 @@ namespace PetriEngine {
 
                 get_variables(e, variables, varPositions, varModifierMap, tuples, false);
             }
+
+        private:
+            bool is_last_variable_expr(const std::shared_ptr<const ColorExpression> expr) const {
```

Comment:
I don't think ownership needs to be transfered here. `const ColerExpression&` (or as a `*`) should be enough. As a minimum send the shared-ptr as a reference.